src/vehicle.cpp

```diff
@@ -6868,23 +6822,26 @@ item vehicle_part::properties_to_item() const
     // this is very lossy.
     float hpofdur = ( float )hp / vpinfo.durability;
     tmp.damage = std::min( 4, std::max<int>( 0, ( 1 - hpofdur ) * 5 ) );
-    // Transfer fuel back to tank, but not to gun or it'll crash
-    if( !tmp.is_gun() && !vpinfo.fuel_type.empty() && vpinfo.fuel_type != "null" && amount > 0 ) {
-        const itype_id &desired_liquid = vpinfo.fuel_type;
-        const int fuel_per_charge = fuel_charges_to_amount_factor( desired_liquid );
-        if( desired_liquid == fuel_type_battery ) {
-            tmp.charges = amount / fuel_per_charge;
-        } else {
-            item liquid( desired_liquid, calendar::turn );
-            liquid.charges = amount / fuel_per_charge;
-            if( liquid.charges > 0 ) {
-                tmp.put_in( liquid );
-            }
-        }
+
+    if( vpinfo.fuel_type == fuel_type_battery ) {
+        tmp.charges = amount;
     }
+
     return tmp;
 }
 
+std::string vehicle_part::name() const {
+    if( base.engine_displacement() > 0 ) {
```

Comment:
Why not just call `item::tname(1, false)` (name without prefix)? This would include things like the corpse name (mountable trophies) and user-defined names (names engraved on the item). Might be too flexible, through. It would erase the redundant code for formatting the displacement/wheel size.